packages/@uppy/dashboard/src/index.js

```diff
@@ -230,6 +235,22 @@ module.exports = class Dashboard extends Plugin {
     })
   }
 
+  openFileEditor = (file) => {
+    const { targets } = this.getPluginState()
+
+    this.setPluginState({
+      showFileEditor: true,
+      activeOverlayType: 'FileEditor'
+    })
+
+    const editors = this._getEditors(targets)
+    editors.forEach((editor) => {
+      const editorPlugin = this.uppy.getPlugin(editor.id)
+      editorPlugin.selectFile(file)
+      setTimeout(editorPlugin.initEditor, 4)
+    })
```

Comment:
❓Questionable way of looping through editors and calling `selectFile` + `initEditor`, with the latter requiring “next tick”.

Reply:
I was thinking this could be done via an event, buuut then it’s a bit “not perfect” that the editor itself emits `file-editor:complete`, but the UI plugin (Dashboard) will have to do `file-editor:start`, passing the file.

Reply:
Could the `selectFile` function return a Promise so we can use `.then()` to call `initEditor` instead of this settimeout?

Do these two function calls need to be separate, or could the `selectFile()` function _also_ set up the editor itself?